src/clp_ffi_py/ir/Query.cpp

```diff
@@ -0,0 +1,25 @@
+#include "Query.hpp"
+
+#include <algorithm>
+#include <string_view>
+
+#include <clp/components/core/src/string_utils.hpp>
+
+namespace clp_ffi_py::ir {
+auto Query::wildcard_matches(std::string_view log_message) const -> bool {
+    if (m_wildcard_list.empty()) {
+        return true;
+    }
+    return std::any_of(
+            m_wildcard_list.begin(),
+            m_wildcard_list.end(),
+            [&](auto const& wildcard_query) {
+                return wildcard_match_unsafe(
```

Comment:
Where will the validation for the input wildcard queries go (in order to use this method)?

Reply:
when parsing the user input in py `__init__`

Reply:
Gotcha, then we should add a line in the constructors to say something like `the wildcard query must be valid (see `wildcard_match_unsafe`).